exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -200,249 +147,206 @@ private static String generateDefaultTaskValue() {
-      return timeSeriesList;
-    }
+    io.opencensus.metrics.export.MetricDescriptor metricDescriptor = metric.getMetricDescriptor();
 
-    // Shared fields for all TimeSeries generated from the same ViewData
+    // Shared fields for all TimeSeries generated from the same Metric
     TimeSeries.Builder shared = TimeSeries.newBuilder();
-    shared.setMetricKind(createMetricKind(view.getWindow(), view.getAggregation()));
+    shared.setMetricKind(createMetricKind(metricDescriptor.getType()));
     shared.setResource(monitoredResource);
-    shared.setValueType(createValueType(view.getAggregation(), view.getMeasure()));
+    shared.setValueType(createValueType(metricDescriptor.getType()));
 
-    // Each entry in AggregationMap will be converted into an independent TimeSeries object
-    for (Entry<List</*@Nullable*/ TagValue>, AggregationData> entry :
-        viewData.getAggregationMap().entrySet()) {
+    // Each entry in timeSeriesList will be converted into an independent TimeSeries object
+    for (io.opencensus.metrics.export.TimeSeries timeSeries : metric.getTimeSeriesList()) {
       TimeSeries.Builder builder = shared.clone();
-      builder.setMetric(createMetric(view, entry.getKey(), domain));
-      builder.addPoints(
-          createPoint(entry.getValue(), viewData.getWindowData(), view.getAggregation()));
+      builder.setMetric(createMetric(metricDescriptor, timeSeries.getLabelValues(), domain));
+
+      io.opencensus.common.Timestamp startTimeStamp = timeSeries.getStartTimestamp();
+      for (io.opencensus.metrics.export.Point point : timeSeries.getPoints()) {
+        builder.addPoints(createPoint(point, startTimeStamp));
+      }
       timeSeriesList.add(builder.build());
     }
-
     return timeSeriesList;
   }
 
-  // Create a Metric using the TagKeys and TagValues.
+  // Create a Metric using the LabelKeys and LabelValues.
   @VisibleForTesting
-  static Metric createMetric(View view, List</*@Nullable*/ TagValue> tagValues, String domain) {
+  static Metric createMetric(
+      io.opencensus.metrics.export.MetricDescriptor metricDescriptor,
+      List<LabelValue> labelValues,
+      String domain) {
     Metric.Builder builder = Metric.newBuilder();
-    // TODO(songya): use pre-defined metrics for canonical views
-    builder.setType(generateType(view.getName().asString(), domain));
+    builder.setType(generateType(metricDescriptor.getName(), domain));
     Map<String, String> stringTagMap = Maps.newHashMap();
-    List<TagKey> columns = view.getColumns();
+    List<LabelKey> labelKeys = metricDescriptor.getLabelKeys();
+
     checkArgument(
-        tagValues.size() == columns.size(), "TagKeys and TagValues don't have same size.");
-    for (int i = 0; i < tagValues.size(); i++) {
-      TagKey key = columns.get(i);
-      TagValue value = tagValues.get(i);
+        labelKeys.size() == labelValues.size(), "LabelKeys and LabelValues don't have same size.");
```

Comment:
I see ambiguity here, if someone using the stats API will get confused by seeing this message.  Can we say "**TagKeys/LabelKeys and TagValues/LabelValues don't have same size.**" @bogdandrutu @songy23

Reply:
Maybe it's less confusing to just call them "key" and "value"? i.e "Keys and Values don't have same size."